src/libdredd/src/mutate_visitor.cc

```diff
@@ -278,6 +279,32 @@ bool MutateVisitor::VisitBinaryOperator(
   return true;
 }
 
+bool MutateVisitor::VisitExpr(clang::Expr* expr) {
+  if (!IsInFunction()) {
+    // Only consider mutating expressions that occur inside functions.
+    return true;
+  }
+
```

Comment:
We need a check that the expression starts and ends in the main source file - see analogous code in the visit methods for unary and binary.

This test case currently leads to a crash:

```
#define E (1, 2, 3)

void foo(int, int, int);

int main() {
  foo E;
}
```

because from Clang's point of view, macros count as not being in the main source file. Perhaps you could add it as a single file test case - I believe the entire `foo E` should get mutated, but the arguments `(1, 2, 3)` should not (as they occur in the macro body).

Reply:
Added a single file test case.